### DISCUSSION TITLE
Support Temperatur/Humiditysensors like DHT21

### DISCUSSION THREAD

**Discussion by lechndo** [emsesp/EMS-ESP32#1299] [Ideas]
Hi guyes, I raelly love EMS-ESP since I fou d it.

Lately I thought that it would be nice to also measure humidity in addition to temperature and startet searching the web for combined  1-wire temperature and humidity sensors.

I found the DHT21 and similar sensors and asked myself if it would me complicated to add support for them.

The sensors are relatively cheap but unfortunately they are not addressable and multiple data pins would be needed for multiple sensors.

Currently I don't have much time but I hope to be able to have a closer look on the matter in the near future.
In the meantime I would be grateful for any ideas and hints...

Best regards,
   Dominik

**Comment by proddy** [comment]
The temperature sensors in EMS-ESP are based on the OneWire protocol, which Dallas sensors support. The DHT11/21/22 although are also single wire sensors they use a private protocol, so unfortunately not compatible.

**Reply by lechndo** [reply]
I thought about some kind of extension to the analog sensor part... because the DHT sensors also require a separate pin per sensor.
The analog sensor part would just need another field to select the sensor type - analog, DHT, etc - to support more none-bus sensors???

**Reply by proddy** [reply]
yes that would be the only way - its more code and not sure it's worth it. Do you have a specific use case for this feature? What are you trying to achieve?

**Reply by lechndo** [reply]
I have my EMS-ESP placed in the basement, connected to my heater.
Actually just temperature and humidity monitoring - maybe with some kind of alarm via HA when the humidity becomes to high in the basement...
And especially during summer we sometimes have humidity issues in the basement...

I know that there are other devices that may have such feature already... but I thought it would be nice to have it on the EMS-ESP because the DHT sensors are relatively cheap and I thought it would be the perfect platform to connect them.

**Reply by MichaelDvP** [reply]
I think i's uncommon to measure the roomtemperature and humidity in the room the heating is mounted and the DHT-one-wire is not intended for longer cable range. The DHT protocol is simple to implement, but the analogsensors needs a lot of changes. They are sorted by one unique GPIO for each sensor, but for DHT we have temperature, humidity and calculated dewpoint on the same gpio. 
Maybe it's better to sort the sensors by name, for most other entities we use a unique name for mqtt/HA. 

I use aqara sensors for temperature and humidity, battery powered zigbee, easy to place in every room and <10€. I've also tested DHT22/AM2302, and BME280 in tasmota devices, but heat from the esp8266 wifi module always gives wrong values and needs correction.

(seems the old round aqaras are sold out and only the expensive temp/humidity/pressure is available. But there is a tuya version on [aliexpress](https://de.aliexpress.com/item/1005004988740044.html?spm=a2g0o.productlist.main.19.3b6443c3DHF0XJ&algo_pvid=9e62022a-2a53-4e93-b5c2-21f737fd7bc7&aem_p4p_detail=202309290425023564995359933500003653210&algo_exp_id=9e62022a-2a53-4e93-b5c2-21f737fd7bc7-9&pdp_npi=4%40dis%21EUR%2110.50%215.89%21%21%2110.85%21%21%40211b88ec16959867027704268ef3ed%2112000033215535204%21sea%21DE%211981526613%21&curPageLogUid=PGuyM4coa0mP&search_p4p_id=202309290425023564995359933500003653210_3))

**Reply by lechndo** [reply]
I didn't want to place the sensor near the heating but somewhere else in the basement and maybe a second one right outside besides my front door... 

As mentioned, I wanted to use these sensors because I haven some laying around and they are rather cheap if I need more.

In contrast, using Zigbee sensors etc., would require to extend my setup (Zigbee HUB etc) quite a lot and I try to avoid that.
I also wanted to have some kind of wired sensor to avoid more batteries, even if Zigbee devices would last quite a while...

Regarding cable length... I found some discussions on the internet where some guys reported to have some sensors working with cables longer than 50m. I remember at least one of them that reported to have the sensor properly working on a 60m UTP cable.
